File: Swift/V2/SfBox.swift
//
//  SfBox.swift
//  Surface Swift
//
//  Created by Webchimp on 26/09/16.
//  Copyright © 2016 AppBuilders. All rights reserved.
//

import Foundation

class SfBox: NSObject {
    
    var top: Float! = 0.0;
    var right: Float! = 0.0;
    var bottom: Float! = 0.0;
    var left: Float! = 0.0;
    
    override init() {
        
        self.top = 0.0;
        self.right = 0.0;
        self.bottom = 0.0;
        self.left = 0.0;
    }
    
    func setBox(top:Float, right:Float, bottom:Float, left:Float) -> Void {
        
        self.top = top;
        self.right = right;
        self.bottom = bottom;
        self.left = left;
    }
    
    func getWidth() -> Float {
        
        return self.right + self.left;
    }
    
    func getHeight() -> Float {
        
        return self.top + self.bottom;
    }
    
    func isEmpty() -> Bool {
        
        if ( self.top == nil && self.right == nil && self.bottom == nil && self.left == nil ) {
            
            return true;
        }
        return false;
    }
    
}

File: Java/SurfaceDemo/surface/src/main/java/com/appbuilders/surface/SfBox.java
package com.appbuilders.surface;

/**
 * Created by ercsanchez on 26/09/17.
 */

public class SfBox extends Object {

    public float top;
    public float right;
    public float bottom;
    public float left;

    public SfBox() {

        this.top = 0;
        this.right = 0;
        this.bottom = 0;
        this.left = 0;
    }

    public SfBox(float top, float right, float bottom, float left) {
        this.setBox(top, right, bottom, left);
    }

    public void setBox(float top, float right, float bottom, float left) {
        this.top = top;
        this.right = right;
        this.bottom = bottom;
        this.left = left;
    }

    public float getWidth() {
        return this.left + this.right;
    }

    public float getHeight() {
        return this.top + this.bottom;
    }

    public boolean isEmpty() {
        return (this.top == 0 && this.right == 0 && this.bottom == 0 && this.left == 0);
    }
}